Simplenote/src/main/java/com/automattic/simplenote/authentication/SignInFragment.kt

```diff
@@ -1,49 +1,137 @@
+import net.openid.appauth.AuthorizationService
+import java.util.UUID
+
+const val TAG = "SignInFragment"
 
 @AndroidEntryPoint
 class SignInFragment: MagicLinkableFragment() {
 
     val viewModel: RequestMagicLinkViewModel by viewModels()
 
-    private var loginWithPassword: TextView? = null
+
+    private var loginWithWordpress: Button? = null
+
+    private var mAuthState: String? = null
+
+    private var signUpCallback: SignUpCallback? = null
+
+    private var authService: AuthorizationService? = null
+
+    private var resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+
+        val data = result.data
+        if (data == null) {
+            Log.d(TAG, "Result code does not match WordPressUtils.OAUTH_ACTIVITY_CODE, or data was null")
+            return@registerForActivityResult
+        }
+
+        val authorizationResponse = AuthorizationResponse.fromIntent(data)
+        val authorizationException = AuthorizationException.fromIntent(data)
+
+        if (authorizationException != null) {
+            val dataUri: Uri = data.data ?: return@registerForActivityResult
+            if (StrUtils.isSameStr(dataUri.getQueryParameter("code"), "1")) {
+                showDialogError(getString(R.string.wpcom_log_in_error_unverified))
+            } else {
+                showDialogError(getString(R.string.wpcom_log_in_error_generic))
+            }
+        } else if (authorizationResponse != null) {
+            // Save token and finish activity.
+            val authSuccess = WordPressUtils.processAuthResponse(
+                activity?.application as Simplenote?,
+                authorizationResponse,
+                mAuthState,
+                true
+            )
+            if (!authSuccess) {
+                showDialogError(getString(R.string.wpcom_log_in_error_generic))
+            } else {
+                AnalyticsTracker.track(
+                    Stat.WPCC_LOGIN_SUCCEEDED,
+                    AnalyticsTracker.CATEGORY_USER,
+                    "wpcc_login_succeeded_signin_activity"
+                )
+                activity?.finish()
+            }
+        }
+    }
 
     override fun inflateLayout(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         val view = inflater.inflate(R.layout.fragment_login, container, false)
-        loginWithPassword = view.findViewById(R.id.login_with_password_button)
-        loginWithPassword?.setOnClickListener { _ ->
-            activity?.let { act ->
-                val intent = Intent(act, SimplenoteCredentialsActivity::class.java)
-                intent.putExtra("EXTRA_IS_LOGIN", true)
-                this.startActivity(intent)
-                act.finish()
+
+        loginWithWordpress = view.findViewById(R.id.button_login_with_wordpress)
+        loginWithWordpress?.setOnClickListener {
+            val ctx = context ?: return@setOnClickListener
```

Comment:
Can the context be null here? If so, should we use a context from the view instead?